src/app/courses-dashboard/store/courses.effect.ts

```diff
@@ -14,11 +16,24 @@ import {
   getCourseByIdSuccess,
   addCourse,
   editCourse,
-  manageCourseSuccess
+  manageCourseSuccess,
+  getAuthors,
+  getAuthorsSuccess
 } from './courses.actions';
 
 @Injectable()
 export class CoursesListEffects {
+  getAuhtors$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(getAuthors),
+      mergeMap(({ searchString }) =>
+        this.authorsService
+          .getAuthors(searchString)
+          .pipe(map((authors: Author[]) => getAuthorsSuccess({ authors })))
```

Comment:
Ideally, in production code you should also handle negative scenarios, so you would have catch operator with getAuthorsFail action, thus, your app could gracefully handle errors or at least log them properly.